packages/react-devtools-shared/src/devtools/views/Components/Components.js

```diff
@@ -17,29 +17,36 @@ import portaledContent from '../portaledContent';
 import {ModalDialog} from '../ModalDialog';
 import SettingsModal from 'react-devtools-shared/src/devtools/views/Settings/SettingsModal';
 import {SettingsModalContextController} from 'react-devtools-shared/src/devtools/views/Settings/SettingsModalContext';
+import {StoreContext} from '../context';
 
 import styles from './Components.css';
 
 function Components(_: {||}) {
+  const {supportsReact} = useContext(StoreContext);
+
   // TODO Flex wrappers below should be user resizable.
   return (
     <SettingsModalContextController>
       <OwnersListContextController>
         <InspectedElementContextController>
-          <div className={styles.Components}>
-            <div className={styles.TreeWrapper}>
-              <Tree />
-            </div>
-            <div className={styles.SelectedElementWrapper}>
-              <NativeStyleContextController>
-                <Suspense fallback={<Loading />}>
-                  <SelectedElement />
-                </Suspense>
-              </NativeStyleContextController>
+          {supportsReact ? (
+            <div className={styles.Components}>
+              <div className={styles.TreeWrapper}>
+                <Tree />
+              </div>
+              <div className={styles.SelectedElementWrapper}>
+                <NativeStyleContextController>
+                  <Suspense fallback={<Loading />}>
+                    <SelectedElement />
+                  </Suspense>
+                </NativeStyleContextController>
+              </div>
+              <ModalDialog />
+              <SettingsModal />
             </div>
-            <ModalDialog />
-            <SettingsModal />
-          </div>
+          ) : (
+            <UnsupportedReactVersion />
```

Comment:
I think we should use the `ModalDialogContext` to show a message if we detect an unsupported React version. We do a similar thing for React Native in [`WarnIfLegacyBackendDetected`](https://github.com/facebook/react/blob/master/packages/react-devtools-shared/src/devtools/views/WarnIfLegacyBackendDetected.js) actually. Conceptually the two are kind of the same.